Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -259,13 +277,56 @@ public class Klaviyo: NSObject  {
                 , "Property values must be of NSString, NSNumber, NSNull, NSDictionary, NSDate, or NSURL. Got: \(String(describing: properties[k as! NSCopying]))")
         }
     }
+}
+
+
+struct KlaviyoSDK {
+    func initialize(with apiKey: String) {
+        dispatchOnMainThread(action: .initialize(apiKey))
+    }
+    
+    func set(profile: Profile) {
+        dispatchOnMainThread(action: .enqueueProile(profile))
+    }
+    
+    func resetProfile() {
+        dispatchOnMainThread(action: .resetProfile)
+    }
     
-    private func dispatchOnMainThread(action: KlaviyoAction) {
-        Task {
-            await MainActor.run {
-                _ = environment.analytics.store.send(action)
-            }
+    func set(email: String) {
+        dispatchOnMainThread(action: .setEmail(email))
+        
+    }
+    
+    func set(externalId: String) {
+        dispatchOnMainThread(action: .setExternalId(externalId))
+    }
+    
+    func set(profileAttribute: String, value: Any) {
+        // This seems tricky to implement with Any - might need to restrict to something equatable, encodable....
+    }
+    
+    func create(event: Event) {
+        dispatchOnMainThread(action: .enqueueEvent(event))
+    }
+    
+    func set(pushToken: Data) {
+        let apnDeviceToken = pushToken.map { String(format: "%02.2hhx", $0) }.joined()
+        dispatchOnMainThread(action: .setPushToken(apnDeviceToken))
+    }
+    
+    func handle(remoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        if let properties = userInfo as? [String: Any], let body = properties["body"] as? [String: Any], let _ = body["_k"] {
```

Comment:
Gotta get better about writing these things down, but didn't we determine we don't want to trigger $opened_push here, only in the UNNofiticationResponse method, as long as they implement both?

Reply:
I also cut a TP ticket about putting de-duping $opened_push by message+push token in that event processor.

Reply:
I think you might want to have both in case you don't implement both options. I don't know why you would do that but having the extra api gives them the option to not have to implement both.

Reply:
ok. as long as we're deduping i guess it doesnt matter.

Reply:
That said potentially we could update the docs though to give options vs implementing both but I thought it wouldn't call both.

Reply:
Ok will re-test!

Reply:
Resolved this!